examples/display_summed_images.py

```diff
@@ -37,58 +37,65 @@ class ImageSumDisplayerTool(Tool):
                     'telgroup': 'ImageSumDisplayerTool.telgroup',
                     'max-events': 'ImageSumDisplayerTool.max_events',
                     'output-suffix': 'ImageSumDisplayerTool.output_suffix'})
-    classes = List([CameraCalibrator,])
+    classes = List([CameraCalibrator,HessioFileReader])
 
     def setup(self):
         # load up the telescope types table (need to first open a file, a bit of
         # a hack until a proper insturment module exists) and select only the
         # telescopes with the same camera type
-        data = next(hessio_event_source(self.infile, max_events=1))
-        data.inst.subarray.info(printer=self.log.info)
-        camtypes = data.inst.subarray.to_table().group_by('camera_type')
+
+        self.reader = HessioFileReader(None, None, input_url=self.infile,
+                                       max_events=self.max_events)
+
+
+        for event in self.reader:
+            camtypes = event.inst.subarray.to_table().group_by('camera_type')
+            event.inst.subarray.info(printer=self.log.info)
+            break
+
         group = camtypes.groups[self.telgroup]
-        self._selected_tels = group['tel_id'].data
+        self._selected_tels = list(group['tel_id'].data)
         self._base_tel = self._selected_tels[0]
         self.log.info("Telescope group %d: %s",
                       self.telgroup,
-                      str(data.inst.subarray.tel[self._selected_tels[0]]))
+                      str(event.inst.subarray.tel[self._selected_tels[0]]))
         self.log.info("SELECTED TELESCOPES:{}".format(self._selected_tels))
         self.calibrator = CameraCalibrator(self.config, self)
+        self.reader.allowed_tels = self._selected_tels
+
 
     def start(self):
         geom = None
         imsum = None
         disp = None
 
-        for data in hessio_event_source(self.infile,
-                                        allowed_tels=self._selected_tels,
-                                        max_events=self.max_events):
-
-            self.calibrator.calibrate(data)
+        for event in self.reader:
+            
+            self.calibrator.calibrate(event)
 
             if geom is None:
-                geom = data.inst.subarray.tel[self._base_tel].camera
+                geom = event.inst.subarray.tel[self._base_tel].camera
                 imsum = np.zeros(shape=geom.pix_x.shape, dtype=np.float)
                 disp = CameraDisplay(geom, title=geom.cam_id)
                 disp.add_colorbar()
                 disp.cmap = 'viridis'
 
-            if len(data.dl0.tels_with_data) <= 2:
+            if len(event.dl0.tels_with_data) <= 2:
                 continue
 
             imsum[:] = 0
-            for telid in data.dl0.tels_with_data:
-                imsum += data.dl1.tel[telid].image[0]
+            for telid in event.dl0.tels_with_data:
+                imsum += event.dl1.tel[telid].image[0]
 
             self.log.info("event={} ntels={} energy={}" \
```

Comment:
According to pep8 line continuation with backslash is discouraged:

this could be reformated as:
```python
self.log.info("event={} ntels={} energy={}".format(
    event.r0.event_id,
    len(event.dl0.tels_with_data),
    event.mc.energy))
```